### PR TITLE
ICP-9242 [KR][eZEX][Smart Gas Vaule Actuator][PROD]Device battery can not change status ,always show 50%.

### PR DESCRIPTION
**Pull request by BYDengS**
@tpmanley and @greens ,this PR is to fix ICP-9242, I have tested it, it can change the battery status "Nomal" to "low" when the battery is low power. Please help review it.

Whats more, there are the reason I add this code.
	description is zone report :: type: 19 value: 0002  //normal power
	description is zone report :: type: 19 value: 000A //low power

### CODE REVIEW

File: devicetypes/smartthings/zigbee-valve.src/zigbee-valve.groovy [hunk 1]
```diff
@@ -91,6 +91,16 @@ def parse(String description) {
             event.value = event.value.toLowerCase()
         }
         sendEvent(event)
+    }else if(description?.startsWith('zone status') || description?.startsWith('zone report')){
+        def splitMsg = description.split(" ");
```
Comment (tpmanley) [inline devicetypes/smartthings/zigbee-valve.src/zigbee-valve.groovy]: We have a helper function for parsing the Zone Status attribute that you should use instead:

```
zs = zigbee.parseZoneStatus(description)
value = zs.isBatterSet() ? 5 : 50
```

File: devicetypes/smartthings/zigbee-valve.src/zigbee-valve.groovy [hunk 2]
```diff
@@ -91,6 +91,16 @@ def parse(String description) {
             event.value = event.value.toLowerCase()
         }
         sendEvent(event)
+    }else if(description?.startsWith('zone status') || description?.startsWith('zone report')){
```
Comment (tpmanley) [inline devicetypes/smartthings/zigbee-valve.src/zigbee-valve.groovy]: add spaces after `}` and `if` and before `{`

**Comment by tpmanley** [comment]
Looks like this change replaces the `else if (descMap.clusterInt == zigbee.IAS_ZONE_CLUSTER && descMap.attrInt == zigbee.ATTRIBUTE_IAS_ZONE_STATUS) {` in `parse` so I think that can be removed.

File: devicetypes/smartthings/zigbee-valve.src/zigbee-valve.groovy [hunk 3]
```diff
@@ -91,6 +91,16 @@ def parse(String description) {
...
+        def splitMsg = description.split(" ");
+        def descValue = Integer.parseInt(splitMsg[-1],16)
+        def value = descValue & 0x0008 ?5:50
+        def result = [:]
+        result.name = 'battery'
+        result.value = value
+        result.descriptionText = "${device.displayName} battery value is ${value}"
+        result.translatable = true
```
Comment (tpmanley) [inline devicetypes/smartthings/zigbee-valve.src/zigbee-valve.groovy]: There are no translations for battery events in the `messages.properties` file so you don't need to set `translatable`.

File: devicetypes/smartthings/zigbee-valve.src/zigbee-valve.groovy [hunk 4]
```diff
@@ -91,6 +91,16 @@ def parse(String description) {
...
         }
         sendEvent(event)
+    }else if(description?.startsWith('zone status') || description?.startsWith('zone report')){
+        def splitMsg = description.split(" ");
+        def descValue = Integer.parseInt(splitMsg[-1],16)
+        def value = descValue & 0x0008 ?5:50
+        def result = [:]
+        result.name = 'battery'
+        result.value = value
+        result.descriptionText = "${device.displayName} battery value is ${value}"
```
Comment (tpmanley) [inline devicetypes/smartthings/zigbee-valve.src/zigbee-valve.groovy]: Use `"${device.displayName} battery was ${result.value}%"`. That's what most DTHs use

**Comment by tpmanley** [comment]
Are you looking to hotfix this change outside of the normal release cycle? If not it should go against `staging` or `master`.

**Comment by BYDengS** [comment]
> Are you looking to hotfix this change outside of the normal release cycle? If not it should go against `staging` or `master`.



> Are you looking to hotfix this change outside of the normal release cycle? If not it should go against `staging` or `master`.

**Comment by BYDengS** [comment]
> Are you looking to hotfix this change outside of the normal release cycle? If not it should go against `staging` or `master`.

our certification team tested this device on prod server, so this issue should be fixed on prod server.

**Comment by BYDengS** [comment]
@tpmanley ,I have modify the code, please review again.

**Comment by varzac** [comment]
It will be on prod server eventually, we typically only target the production branch if it's a hotfix that needs to go out immediately.

**Comment by tpmanley** [comment]
My plan is to get merge this into `staging` if QA can verify it first and then it can go to prod on the normal schedule (tomorrow).

**Comment by varzac** [comment]
This is locally running, but it looks like this message should be coming to the cloud, but it would probably be nice to handle locally at some point.  No need for that to hold anything up though.

**Review by varzac — APPROVED** [review]

**Review by tpmanley — APPROVED** [review]
Will merge once it passes QA

**Pull request merged**